scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5240 @@
+#   speed things up. If it's not needed, it may be about twice as slow. I think
+#   it should depend on the accuracy setting.
+#  in tests, check reference value against that produced using np.vectorize?
+#  add `axis` to `ks_1samp`
+#  Getting `default_rng` takes forever! OK to do it only when support is called?
+#  User tips for faster execution:
+#  - pass NumPy arrays
+#  - pass inputs of floating point type (not integers)
+#  - prefer NumPy scalars or 0d arrays over other size 1 arrays
+#  - pass no invalid parameters and disable invalid parameter checks with iv_profile
+#  - provide a Generator if you're going to do sampling
+#  add options for drawing parameters: log-spacing
+#  accuracy benchmark suite
+#  Should caches be attributes so we can more easily ensure that they are not
+#   modified when caching is turned off?
+#  Make ShiftedScaledDistribution more efficient - only process underlying
+#   distribution parameters as necessary.
+#  Reconsider `all_inclusive`
+#  Should process_parameters update kwargs rather than returning? Should we
+#   update parameters rather than setting to what process_parameters returns?
+
+# Questions:
+# 1.  I override `__getattr__` so that distribution parameters can be read as
+#     attributes. We don't want uses to try to change them.
+#     - To prevent replacements (dist.a = b), I could override `__setattr__`.
+#     - To prevent in-place modifications, `__getattr__` could return a copy,
+#       or it could set the WRITEABLE flag of the array to false.
+#     Which should I do?
+# 2.  `cache_policy` is supported in several methods where I imagine it being
+#     useful, but it needs to be tested. Before doing that:
+#     - What should the default value be?
+#     - What should the other values be?
+#     Or should we just eliminate this policy?
+# 3.  `validation_policy` is supported in a few places, but it should be checked for
+#     consistency. I have the same questions as for `cache_policy`.
+# 4.  `tol` is currently notional. I think there needs to be way to set
+#     separate `atol` and `rtol`. Some ways I imagine it being used:
+#     - Values can be passed to iterative functions (quadrature, root-finder).
+#     - To control which "method" of a distribution function is used. For
+#       example, if `atol` is set to `1e-12`, it may be acceptable to compute
+#       the complementary CDF as 1 - CDF even when CDF is nearly 1; otherwise,
+#       a (potentially more time-consuming) method would need to be used.
+#     I'm looking for unified suggestions for the interface, not ad hoc ideas
+#     for using tolerances. Suppose the user wants to have more control over
+#     the tolerances used for each method - how do they specify it? It would
+#     probably be easiest for the user if they could pass tolerances into each
+#     method, but it's easiest for us if they can only set it as a property of
+#     the class. Perhaps a dictionary of tolerance settings?
```

Comment:
Always accept `tol`. In the case that `rtol` and/or `atol` also are accepted, but a `tol` is passed, then set `rtol = rtol ?? tol ?? DEFAULT_TOL` (I use `??` here as an `if not None` or `if _ not in kwargs` operator, i.e. a stricter `or`).